src/date-picker/calendar/__tests__/calendar.test.tsx

```diff
@@ -392,14 +440,14 @@ describe('Date picker calendar', () => {
         expect(findCalendarHeaderText(wrapper)).toBe('March 2018');
       });
 
-      test('should display next month if there is no enabled date in the current', () => {
+      test('should display the current month if there is no enabled date in the current', () => {
         const dateLimit = new Date(2018, 3, 10).getTime();
         const isDateEnabled = (date: Date) => date.getTime() > dateLimit;
-        const { wrapper } = renderDatePicker({ ...defaultProps, isDateEnabled });
+        const { wrapper } = renderDatePicker({ ...defaultProps, value: '2018-03-21', isDateEnabled });
         act(() => wrapper.findOpenCalendarButton().click());
-        act(() => wrapper.setInputValue('2018/03/21'));
-        expect(findCalendarHeaderText(wrapper)).toBe('April 2018');
-        expect(findFocusableDateText(wrapper)).toBe('11');
+        expect(findCalendarHeaderText(wrapper)).toBe('March 2018');
+        expect(wrapper.findCalendar()!.findSelectedDate().getElement().textContent).toBe('21');
+        expect(findFocusableDateText(wrapper)).toBeNull();
```

Comment:
In the `getBaseDate` function, we now return the first day of the month if no other day is selected, and it is my understanding from the comment there that this day would be used as the focus target. Doesn't this conflict with our `toBeNull` expectation here?

Reply:
In that case no day can be focused.

Reply:
What @pan-kot said. In this test case, March does not contain any focusable / enabled day. It marks 2018-03-21, the input's value, as selected but it's not enabled/focusable.